src/test/java/streams/part1/exercise/Exercise2.java

```diff
@@ -0,0 +1,116 @@
+package streams.part1.exercise;
+
+import lambda.data.Employee;
+import lambda.data.JobHistoryEntry;
+import lambda.data.Person;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@SuppressWarnings({"ConstantConditions", "unused"})
+class Exercise2 {
+
+    @Test
+    void calcAverageAgeOfEmployees() {
+        List<Employee> employees = getEmployees();
+
+        Double expected = employees.stream()
+                .mapToDouble(employee -> employee.getPerson().getAge())
+                .average().orElse(-1);
+
+        assertThat(expected, Matchers.closeTo(33.66, 0.1));
+    }
+
+    @Test
+    void findPersonWithLongestFullName() {
+        List<Employee> employees = getEmployees();
+
+        Person expected = employees.stream()
+                .map(Employee::getPerson)
+                .max(Comparator.comparingInt(p -> p.getFullName().length()))
+                .orElseThrow(UnsupportedOperationException::new);
+
+        assertThat(expected, Matchers.is(employees.get(1).getPerson()));
+    }
+
+    @Test
+    void findEmployeeWithMaximumDurationAtOnePosition() {
```

Comment:
I wrote the same realization for this task, but then read the name carefully and realized, that it must work in another way - you should summurize "duration" for each "position" of each Employee and return the one, who has maximum "duration" (for example, you can add `new JobHistoryEntry(1, "dev", "google")` 10 times for the  `employees.get(0)` and your `Employee expected` must return him, but you current solution won't do that).